src/custom/state/swap/extension.ts

```diff
@@ -47,6 +53,7 @@ export function extendExactOutTrade(params: Pick<ExtendedTradeParams, 'exactOutT
   return {
     ...exactOutTrade,
     inputAmount: inputAmountWithFee,
+    inputAmountWithFee,
```

Comment:
why are inputAmount and inputAmountWithFee equal?

Reply:
it's for types. The extended TradeWithFee interface required `inputAmountWithFee` the reason `inputAmount` is === `inputAmountWithFee` is that we are adding the fee to the UNI calculated `inputAmount` and we want to show that in the interface in the input field to the user

Reply:
and this allows us to not change anything in the Swap component

Reply:
Aha, ok, so it's a hack. Cause the right approach is to have the "parts" (fee, amount before fee, ..) and decide in each place how we display it or use it. However, I see your point on doing the hack allows us to not change their original code

Maybe add a comment about the hack then?

Reply:
done